lms/resources/_js_config.py

```diff
@@ -367,3 +368,24 @@ def _hypothesis_client(self):
                 }
             ]
         }
+
+    def _sync_api(self):
+        if not self._context.should_use_section_groups:
+            return None
```

Comment:
It's gonna be up to the backend to decide when the sync API should be called and when not. The backend will either put the sync API settings in the JS config or not, and the frontend will just call the sync API if the settings are present